src/Endpoints/Delegates/HandlesTasks.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MeiliSearch\Endpoints\Delegates;
+
+use MeiliSearch\Endpoints\Tasks;
+
+/**
+ * @property Tasks tasks
+ */
+trait HandlesTasks
+{
+    public function getTask($uid): array
+    {
+        return $this->tasks->get($uid);
+    }
+
+    public function getAllTasks(): array
+    {
+        return $this->tasks->all();
+    }
+
+    public function waitForTask($uid, $timeoutInMs = 5000, $intervalInMs = 50): array
```

Comment:
maybe the parameter should be called `taskUid` to be sure that it is not confused with `indexUid`